spring/src/main/java/uk/ac/bbsrc/tgac/miso/spring/ajax/ExternalSectionControllerHelperService.java

```diff
@@ -273,11 +273,11 @@ public JSONObject listRunsDataTable(HttpSession session, JSONObject json) {
               if (spc.getPartitions().size() > 0) {
                 for (Partition spp : spc.getPartitions()) {
                   if (spp.getPool() != null) {
-                    if (spp.getPool().getPoolableElementViews().size() > 0) {
-                      for (PoolableElementView dilution : spp.getPool().getPoolableElementViews()) {
-                        if (dilution.getProjectId().equals(projectId)) {
+                    if (spp.getPool().getPoolDilutions().size() > 0) {
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Merge this if statement with the enclosing one. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1066)
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Use isEmpty() to check whether the collection is empty or not. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1155)